### PR TITLE
Feature/4257/check url

### PR DESCRIPTION
**Pull request by garyluu**
Initial PR for the checkURL for dockstore/dockstore#4257

Still considering whether to change input to an array of files or not. Leaving it as a single file for now.
Haven't tested it with dockstore/dockstore yet, though it should be easily configuration by changing the template.yaml
Tested it with SAM CLI, seems to work and return the message and status

```
curl http://127.0.0.1:3000/lambda?url=https://www.google.ca
{"message":true}
```

Also ran eslint --fix on every js file.
Simplied the eslinting commands in the github actions
Had to install the packages from the lambda in CI which means:
- devDependencies (testing dependencies) should be installed in the root package.json
- lambda dependencies (to actually run the lambda) should be install in the specific lambda's package.json

Edit: seems i should've made the input a query parameter instead. It's not often that GET requests have body

### CODE REVIEW

File: checkUrlExists/lambda/app.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// const axios = require('axios')
+// const url = 'http://checkip.amazonaws.com/';
+
+const fetch = require("node-fetch");
+/**
+ * TODO: Change to array of URLs to parse
+ * Always returns 200. Body is true if file URL is valid, body is false if file URL is not valid or something has gone wrong
+ * The request is expected to have a body that is the just the plan text URL to check
```
Comment (coverbeck) [inline checkUrlExists/lambda/app.js:8]: plain

**Review by coverbeck — APPROVED** [review]
What are we going to do for non-http urls (s3, etc.)? It's the responsibility of the web service to not call this lambda in that case?

**Comment by garyluu** [comment]
> What are we going to do for non-http urls (s3, etc.)? It's the responsibility of the web service to not call this lambda in that case?

Ya, I was thinking the web service would do as much as possible. Everything except checking the file.

File: checkUrlExists/lambda/app.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// const axios = require('axios')
```
Comment (denis-yuen) [inline checkUrlExists/lambda/app.js:1]: What were these two lines for?

File: checkUrlExists/template.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  # https://github.com/awslabs/serverless-application-model/blob/master/docs/internals/generated_resources.rst#api
+  LambdaApi:
+    Description: "API Gateway endpoint URL for Prod stage for Hello World function"
+    Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/hello/"
+  LambdaFunction:
+    Description: "Hello World Lambda Function ARN"
+    Value: !GetAtt LambdaFunction.Arn
+  LambdaFunctionIamRole:
+    Description: "Implicit IAM Role created for Hello World function"
+    Value: !GetAtt LambdaFunctionRole.Arn
```
Comment (denis-yuen) [inline checkUrlExists/template.yaml:39]: Some hello world boilerplate

**Comment by garyluu** [comment]
Holding off on reviews, the GET with a body is bothering me more than I thought, even for a first pass

**Review by denis-yuen — APPROVED** [review]
> Still considering whether to change input to an array of files or not. Leaving it as a single file for now. 

Think it would be nice to have it be one lambda call per workflow version in general

> seems i should've made the input a query parameter instead. It's not often that GET requests have body

Would make it easier to read what we're checking in the logs

File: checkUrlExists/lambda/app.js (hunk 2 above)
Comment (garyluu) [reply checkUrlExists/lambda/app.js:1]: The new lambda was generated with SAM CLI...i have quite a bit of default stuff left in

**Review by NatalieEO — APPROVED** [review]

**Pull request merged**